test/cli-tests/warn_if_no_write_permissions_test.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+. ./test/helper.sh
+
+test_install_dir="$test_fixtures_dir/no_write_permissions_test"
```

Comment:
Maybe to make this test more explicit, you could add another variable called `non_writable_parent_dir="$test_fixtures_dir/..."` and set `test_install_dir="$non_writable_parent_dir/ruby"`.